### PR TITLE
Add property logCategory

### LINKED ISSUE #15: Add log category to UnityTask
The logs for each tasks are based on the taskname. It would be nice to add another level so you can invoke gradle multiple times with different parameters without overriding the logs and reports. This is mainly for a jenkins workflow.

The default should be `""`. There should be the 3 typical ways to set/override it.
* property in task
* property in extension
* gradle parameter `-P...=`

### PR DESCRIPTION
**Pull request by marcolink**
## Description
The logs for each tasks are based on the taskname. It would be nice to add another level so you can invoke gradle multiple times with different parameters without overriding the logs and reports. This is mainly for a jenkins workflow.

The default should be `""`. There should be the 3 typical ways to set/override it.
* property in task
* property in extension
* gradle parameter `-P...=`

resolves #15 

## Changes
* ![ADD] logCategory 

[NEW]:https://atlas-resources.wooga.com/icons/icon_new.svg "New"
[ADD]:http://resources.atlas.wooga.com/icons/icon_add.svg "Add"
[IMPROVE]:http://resources.atlas.wooga.com/icons/icon_improve.svg "IMPROVE"
[CHANGE]:http://resources.atlas.wooga.com/icons/icon_change.svg "Change"
[FIX]:http://resources.atlas.wooga.com/icons/icon_fix.svg "Fix"
[UPDATE]:http://resources.atlas.wooga.com/icons/icon_update.svg "Update"

[BREAK]:http://resources.atlas.wooga.com/icons/icon_break.svg "Remove"
[REMOVE]:http://resources.atlas.wooga.com/icons/icon_remove.svg "Remove"
[IOS]:http://resources.atlas.wooga.com/icons/icon_iOS.svg "iOS"
[ANDROID]:http://resources.atlas.wooga.com/icons/icon_android.svg "Android"
[WEBGL]:http://resources.atlas.wooga.com/icons/icon_webGL.svg "Web:GL"

### CODE REVIEW

File: src/integrationTest/groovy/wooga/gradle/unity/BaseBatchModeIntegrationSpec.groovy [hunk 1]
```diff
@@ -179,6 +176,43 @@ class BaseBatchModeIntegrationSpec extends UnityIntegrationSpec {
...
+        and:
+        if (useGetter) {
+            buildFile << """
+            mUnity.setLogCategory("${value}")
+        """.stripIndent()
+        } else {
+            buildFile << """
+            mUnity.logCategory = "${value}"
+        """.stripIndent()
+        }
```
Comment (Larusso) [inline src/integrationTest/groovy/wooga/gradle/unity/BaseBatchModeIntegrationSpec.groovy]: I would generate the method call. A `mUnity.logCategory = ...` is the same as `mUnity.setLogCategory(...)`
Check this test

https://github.com/wooga/atlas-unity/blob/b5fd4031f5bf44bcbc75da86b9cfb0fdbfaa2ef0/src/integrationTest/groovy/wooga/gradle/unity/UnityBatchModeIntegrationSpec.groovy#L21-L53

File: src/main/groovy/wooga/gradle/unity/tasks/AbstractUnityTask.groovy [hunk 2]
```diff
@@ -98,6 +105,20 @@ abstract class AbstractUnityTask<T extends AbstractUnityTask> extends Convention
         this.retrieveAction().conventionMapping
     }
 
+
+
+    /*
+    private def getLogCategory() {
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/tasks/AbstractUnityTask.groovy]: Old code?

File: src/main/groovy/wooga/gradle/unity/tasks/UnityPackage.groovy [hunk 3]
```diff
@@ -142,12 +141,19 @@ class UnityPackage extends AbstractBatchModeTask {
         inputFiles(project.files([source]))
     }
 
+    @Override
+    ConventionMapping getConventionMapping() {
+        return this
+    }
+
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/tasks/UnityPackage.groovy]: This should not be here

File: src/integrationTest/groovy/wooga/gradle/unity/BaseBatchModeIntegrationSpec.groovy [hunk 4]
```diff
@@ -208,7 +235,9 @@ class BatchModeIntegrationSpec extends IntegrationSpec {
     }
 
     @Unroll
-    @IgnoreIf({ System.getProperty("os.name").toLowerCase().contains("windows") })
+    @Ignore
+    //test occasionally fails on jenkins
```
Comment (Larusso) [inline src/integrationTest/groovy/wooga/gradle/unity/BaseBatchModeIntegrationSpec.groovy]: FYI
`@Ignore` supports a reason message
http://spockframework.org/spock/docs/1.1/all_in_one.html#_ignore

`@Ignore("test occasionally fails on jenkins")`

File: src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy [hunk 5]
```diff
@@ -209,7 +216,7 @@ class DefaultUnityPluginExtension implements UnityPluginExtension {
 
     @Override
     DefaultUnityPluginExtension unityPath(Object path) {
-        customUnityPath = fileResolver.resolveLater(path)
+        this.setUnityPath(path)
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy]: I still don't get why this change is in this pull request

File: src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy [hunk 6]
```diff
@@ -335,6 +358,7 @@ class DefaultUnityPluginExtension implements UnityPluginExtension {
     @Override
     ExecResult activate(Action<? super ActivationSpec> action) {
         ActivationAction activationAction = activationActionFactory.create()
+        //TODO: configure action
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy]: I think this TODO is solved

File: src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy [hunk 7]
```diff
@@ -118,14 +118,15 @@ class UnityPlugin implements Plugin<Project> {
...
-                    project.file("${project.buildDir}/logs/${task.name}.log")
+                    project.file("${project.buildDir}/logs/${task.getLogCategory()}/${task.name}.log")
                 }
             }
         })
     }
 
     static void applyBaseConvention(ConventionMapping taskConventionMapping, UnityPluginExtension extension) {
         taskConventionMapping.map "unityPath", { extension.unityPath }
+        taskConventionMapping.logCategory = { extension.logCategory }
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy]: We should check if we need to wrap values in closures with this mapping style.

File: src/main/groovy/wooga/gradle/unity/UnityPluginExtension.groovy [hunk 8]
```diff
@@ -24,14 +24,14 @@ import wooga.gradle.unity.batchMode.*
 
 interface UnityPluginExtension extends UnityPluginTestExtension {
 
-    File getUnityPath()
-
     void setUnityPath(Object path)
 
-    File getUnityLicenseDirectory()
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/UnityPluginExtension.groovy]: Why are these properties flipped?

File: src/main/groovy/wooga/gradle/unity/batchMode/BaseBatchModeSpec.groovy [hunk 9]
```diff
@@ -17,9 +17,7 @@
 
 package wooga.gradle.unity.batchMode
 
-import org.gradle.api.internal.IConventionAware
-
-interface BaseBatchModeSpec extends IConventionAware {
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/batchMode/BaseBatchModeSpec.groovy]: No longer needed? I thought we just added this here.

File: src/main/groovy/wooga/gradle/unity/tasks/AbstractUnityTask.groovy [hunk 10]
```diff
@@ -98,6 +104,8 @@ abstract class AbstractUnityTask<T extends AbstractUnityTask> extends Convention
         this.retrieveAction().conventionMapping
     }
 
+    @Optional
+    @Input
```
Comment (Larusso) [inline src/main/groovy/wooga/gradle/unity/tasks/AbstractUnityTask.groovy]: No this method is not a gradle Input.

File: src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy (hunk 5 above)
Comment (marcolink) [reply src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy]: This is not changing any behavior, I just wanted to streamline the setter implementation for all properties in this class.

File: src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy (hunk 7 above)
Comment (marcolink) [reply src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy]: It is needed:

```java
public void propertyMissing(String name, Object value) {
        if (value instanceof Closure) {
            map(name, (Closure) value);
        } else {
            throw new MissingPropertyException(name, getClass());
        }
    }
```

File: src/main/groovy/wooga/gradle/unity/UnityPluginExtension.groovy (hunk 8 above)
Comment (marcolink) [reply src/main/groovy/wooga/gradle/unity/UnityPluginExtension.groovy]: i think it's ok since we don't provide a setter

File: src/main/groovy/wooga/gradle/unity/batchMode/BaseBatchModeSpec.groovy (hunk 9 above)
Comment (marcolink) [reply src/main/groovy/wooga/gradle/unity/batchMode/BaseBatchModeSpec.groovy]: will roll it back!

File: src/main/groovy/wooga/gradle/unity/tasks/AbstractUnityTask.groovy (hunk 10 above)
Comment (marcolink) [reply src/main/groovy/wooga/gradle/unity/tasks/AbstractUnityTask.groovy]: rebase madness

File: src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy (hunk 5 above)
Comment (Larusso) [reply src/main/groovy/wooga/gradle/unity/DefaultUnityPluginExtension.groovy]: Ah I had code duplication! Ok
